Translate "Uncontrolled components" page (#173)
reactjs/pl.reactjs.org: content/docs/uncontrolled-components.md

---
id: uncontrolled-components
title: Komponenty niekontrolowane
permalink: docs/uncontrolled-components.html
---

W większości przypadków zalecamy stosowanie [komponentów kontrolowanych](/docs/forms.html#controlled-components) do implementacji formularzy. W komponencie kontrolowanym, dane formularza są obsługiwane przez komponent reactowy. Alternatywą są komponenty niekontrolowane, w których dane formularza są obsługiwane przez sam DOM.

> Aby stworzyć komponent niekontrolowany, zamiast pisać funkcję obsługującą każdą zmianę stanu, możesz [użyć polecenia ref](/docs/refs-and-the-dom.html), aby uzyskać wartości formularza z DOM.

Reviewer: ```suggestion
Aby stworzyć komponent niekontrolowany, zamiast pisać funkcję obsługującą każdą zmianę stanu, możesz [użyć właściwości ref](/docs/refs-and-the-dom.html), aby uzyskać wartości formularza z DOM.
```
`ref` jest właściwością (prop).

> [**Try it on CodePen**](https://codepen.io/gaearon/pen/WooRWa?editors=0010)

Reviewer: ```suggestion
[**Przetestuj kod na CodePen**](https://codepen.io/gaearon/pen/WooRWa?editors=0010)
```

> Ponieważ komponent niekontrolowany zachowuje źródło prawdy w DOM, czasami łatwiej jest zintegrować kod reactowy z kodem niereactowym, gdy używa się komponentów niekontrolowanych. Dzięki temu może pojawić się również nieco mniej kodu, jeśli chcesz podejść do tematu z grubsza. W przeciwnym razie zwykle powinieneś używać komponentów kontrolowanych.

Reviewer: ```suggestion
Ponieważ komponent niekontrolowany zachowuje źródło prawdy w DOM, czasami łatwiej jest zintegrować kod reactowy z kodem niereactowym, gdy używa się komponentów niekontrolowanych. Wpłynie to również na mniejszą ilość kodu i pozwoli na wprowadzenie zmian "na szybko". W przeciwnym razie zwykle powinno się używać komponentów kontrolowanych.
```
1. Zdanie pośrodku brzmi wg mnie dziwnie. Moje też nie najlepiej, szczerze powiedziawszy, ale chyba udało mi się wyznaczyć kierunek ;) Zastanów się, jak to lepiej napisać.
2. W dokumentacji piszemy bezosobowo.

Author: 1. Rozumiem, to trochę pomęczymy to zdanie wspólnie 
```suggestion
Wpłynie to również na mniejszą ilość kodu **i pozwoli na wprowadzenie zmian "na szybko"**
```
W oryginalnej wersji jest "It can also be slightly less code **if you want to be quick and dirty**"

Męczę to zdanie i męczę i wydaje mi się, że trzeba je całkiem przekształcić. W oryginale czuć, że to zdanie jest nacechowane negatywnie przez zwrot "quick and dirty" - myślę, że w naszej wersji możemy pominąć ten pejoratywny ładunek. Jeśli mamy używać bezokoliczników, to utrzymanie tego negatywnego ładunku jest zadaniem ponad moje możliwości :-P

Co do Twojej wersji (i mojej poprzedniej) określmy wyraźniej podmiot, zamiast używania zaimka "to", bo ciężko było mi załapać treść zdania. Trudno też było mi zrozumieć, dlaczego użycie komponentów niekontrolowanych, pozwoli na wprowadzenie zmian "na szybko".

Moja dalsza propozycja:

Użycie komponentów niekontrolowanych pozwala zaimplementować formularz "na szybko" dzięki mniejszej ilości kodu.

Author: 1. Rozumiem, to trochę pomęczymy to zdanie wspólnie 
```suggestion
Wpłynie to również na mniejszą ilość kodu **i pozwoli na wprowadzenie zmian "na szybko"**
```
W oryginalnej wersji jest "It can also be slightly less code **if you want to be quick and dirty**"

Męczę to zdanie i męczę i wydaje mi się, że trzeba je całkiem przekształcić. W oryginale czuć, że to zdanie jest nacechowane negatywnie przez zwrot "quick and dirty" - myślę, że w naszej wersji możemy pominąć ten pejoratywny ładunek. Jeśli mamy używać bezokoliczników, to utrzymanie tego negatywnego ładunku jest zadaniem ponad moje możliwości :-P

Co do Twojej wersji (i mojej poprzedniej) określmy wyraźniej podmiot, zamiast używania zaimka "to", bo ciężko było mi załapać treść zdania. Trudno też było mi zrozumieć, dlaczego użycie komponentów niekontrolowanych, pozwoli na wprowadzenie zmian "na szybko".

Dodatkowo ostatnie zdanie "W przeciwnym razie..." traci sens, bo nie odwołujemy się już do "if you want to be quick and dirty".

Poniżej moja dalsza propozycja:

```suggestion
Ponieważ komponent niekontrolowany zachowuje źródło prawdy w DOM, czasami łatwiej jest zintegrować kod reactowy z kodem niereactowym, gdy używa się komponentów niekontrolowanych. Użycie komponentów niekontrolowanych pozwala zaimplementować formularz "na szybko" dzięki mniejszej ilości kodu. Zwykle jednak powinno używać się komponentów kontrolowanych.
```

Reviewer: Druga wersja mi pasuje. Wrzucaj! :)

> Jeśli nadal nie jest jasne, jakiego rodzaju komponentu należy użyć w konkretnej sytuacji, pomocny może okazać się [ten artykuł o kontrolowanych i niekontrolowanych danych wejściowych](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/).

Reviewer: ```suggestion
Jeśli nadal nie jest jasne, jakiego rodzaju komponentu należy użyć w konkretnej sytuacji, pomocny może okazać się [ten artykuł o kontrolowanych i niekontrolowanych polach formularza](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/).
```

### Wartości domyślne {#default-values}

> W reactowym cyklu życia renderowania wartość atrybutu `value` przypisanego do elementów formularza zastąpi wartość w DOM. W przypadku komponentu niekontrolowanego często chcesz, aby React określił wartość początkową, ale kolejne aktualizacje pozostają niekontrolowane. Aby obsłużyć ten przypadek, zamiast atrybutu `value` można podać atrybut `defaultValue`.

Reviewer: ```suggestion
W reactowym cyklu życia renderowania, wartość atrybutu `value` przypisanego do elementów formularza zastąpi wartość w DOM. W przypadku komponentu niekontrolowanego często zależy nam, aby to React określił wartość początkową, ale by kolejne aktualizacje pozostały niekontrolowane. Aby obsłużyć ten przypadek, zamiast atrybutu `value` można podać atrybut `defaultValue`.
```

```javascript{7}
render() {
  return (
    <form onSubmit={this.handleSubmit}>
      <label>
        Imię:
        <input
          defaultValue="Bob"
          type="text"
          ref={this.input} />
      </label>
      <input type="submit" value="Wyślij" />
    </form>
  );
}
```

> Podobnie, `<input type="checkbox">` i `<input type="radio">` obsługują `defaultChecked`, a `<select>` i `<textarea>` obsługują `defaultValue`.

Reviewer: ```suggestion
Podobnie `<input type="checkbox">` i `<input type="radio">` obsługują atrybut `defaultChecked`, a `<select>` i `<textarea>` obsługują `defaultValue`.
```

## Znacznik `input` dla plików {#the-file-input-tag}

> W HTML `<input type="file">` pozwala użytkownikowi wybrać jeden lub więcej plików z pamięci urządzenia, które mają zostać przesłane na serwer lub obsłużone przez JavaScript przez [File API](https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications).

Reviewer: ```suggestion
W HTML-u `<input type="file">` pozwala użytkownikowi wybrać z pamięci urządzenia jeden lub więcej plików, które mają zostać przesłane na serwer lub obsłużone przez JavaScript za pomocą [interfejsu API dla plików](https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications).
```
